## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :x: [more...](#16215f3c-e4f3-4940-8044-f6fab7dbc225) |
| mint-gateway-azure.sh | :x: [more...](#7383e490-4324-4316-b1fb-d9f0bce6a9e3) |
<a name="16215f3c-e4f3-4940-8044-f6fab7dbc225"></a>
### 5433-adbe83f/mint-gateway-s3.sh.log:
```
Running with
SERVER_ENDPOINT: minikube:30508
ACCESS_KEY:      AKIAJH7GQKYNU3C2ADXA
SECRET_KEY:      ***REDACTED***
ENABLE_HTTPS:    0
SERVER_REGION:   us-east-1
MINT_DATA_DIR:   /mint/data
MINT_MODE:       core

To get logs, run 'docker cp 16226f3aef3f:/mint/log /tmp/mint-logs'
(1/11) Running aws-sdk-go tests ... done in 2 seconds
(2/11) Running aws-sdk-php tests ... done in 1 minutes and 23 seconds
(3/11) Running aws-sdk-ruby tests ... done in 1 minutes and 45 seconds
(4/11) Running awscli tests ... done in 8 minutes and 26 seconds
(5/11) Running mc tests ... done in 6 minutes and 28 seconds
(6/11) Running minio-dotnet tests ... done in 1 minutes and 39 seconds
(7/11) Running minio-go tests ... done in 24 seconds
(8/11) Running minio-java tests ... FAILED in 52 seconds
{
  "name": "minio-java",
  "function": "setBucketPolicy(String bucketName, String objectPrefix, PolicyType policyType)",
  "duration": 6374,
  "status": "FAIL",
  "error": "error occured\nErrorResponse(code=AccessDenied, message=Access Denied., bucketName=null, objectName=null, resource=/minio-java-test-3j3b39/set-bucket-policy-read-write/minio-java-test-j2so6e, requestId=3L137, hostId=3L137)\nrequest={method=GET, url=http://minikube:30508/minio-java-test-3j3b39/set-bucket-policy-read-write/minio-java-test-j2so6e, headers=Host: minikube:30508\nUser-Agent: Minio (amd64; amd64) minio-java/dev\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20180120T010207Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20180120/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=403, headers=Accept-Ranges: bytes\nContent-Type: application/xml\nServer: Minio/DEVELOPMENT.2018-01-20T00-37-36Z (linux; amd64)\nVary: Origin\nX-Amz-Request-Id: 150B5EF9FFFCB45D\nDate: Sat, 20 Jan 2018 01:02:07 GMT\nTransfer-Encoding: chunked\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:1205), io.minio.MinioClient.executeGet(MinioClient.java:1298), io.minio.MinioClient.getObject(MinioClient.java:1606), io.minio.MinioClient.getObject(MinioClient.java:1510), FunctionalTest.testSetBucketPolicy(FunctionalTest.java:2427), FunctionalTest.setBucketPolicy_test4(FunctionalTest.java:2482), FunctionalTest.runQuickTests(FunctionalTest.java:2788), FunctionalTest.main(FunctionalTest.java:2830)]"
}

Executed 7 out of 11 tests successfully.
```
<a name="7383e490-4324-4316-b1fb-d9f0bce6a9e3"></a>
### 5433-adbe83f/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT: minikube:30518
ACCESS_KEY:      minioazure1
SECRET_KEY:      ***REDACTED***
ENABLE_HTTPS:    0
SERVER_REGION:   us-east-1
MINT_DATA_DIR:   /mint/data
MINT_MODE:       core

To get logs, run 'docker cp 108901328a24:/mint/log /tmp/mint-logs'
(1/11) Running aws-sdk-go tests ... done in 1 seconds
(2/11) Running aws-sdk-php tests ... done in 2 minutes and 54 seconds
(3/11) Running aws-sdk-ruby tests ... done in 1 minutes and 2 seconds
(4/11) Running awscli tests ... done in 7 minutes and 47 seconds
(5/11) Running mc tests ... done in 6 minutes and 53 seconds
(6/11) Running minio-dotnet tests ... done in 55 seconds
(7/11) Running minio-go tests ... done in 2 seconds
(8/11) Running minio-java tests ... done in 38 seconds
(9/11) Running minio-js tests ... done in 1 minutes and 56 seconds
(10/11) Running minio-py tests ... FAILED in 1 minutes and 1 seconds
{
  "duration": 1262,
  "args": {
    "object_name": "1b6e513a-c6be-4b6f-b656-7a54c54e3fc6",
    "bucket_name": "minio-py-test-8ea9f45c-923d-4328-8d3e-dd08fd23015d"
  },
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1127, in test_list_incomplete_uploads\n    raise ValueError(\"Listed upload ids do not match the created ids\")\nValueError: Listed upload ids do not match the created ids\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1807, in main\n    test_list_incomplete_uploads(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1129, in test_list_incomplete_uploads\n    raise Exception(err)\nException: Listed upload ids do not match the created ids\n",
  "status": "FAIL",
  "function": "list_incomplete_uploads(bucket_name, prefix, recursive)",
  "name": "minio-py:test_list_incomplete_uploads",
  "message": "Listed upload ids do not match the created ids"
}

Executed 9 out of 11 tests successfully.
```

--- a/pkg/madmin/README.md
+++ b/pkg/madmin/README.md
@@ -1,7 +1,7 @@
 # Minio Admin Library. [![Slack](https://slack.minio.io/slack?type=svg)](https://slack.minio.io)
 The Minio Admin Golang Client SDK provides APIs to manage Minio services.
 
-This quickstart guide will show you how to install the Minio Admin client SDK, connect to Minio admin service, and provide a walkthrough of a simple file uploader. 
+This quickstart guide will show you how to install the Minio Admin client SDK, connect to Minio admin service, and provide a walkthrough of a simple file uploader.
 
 This document assumes that you have a working [Golang setup](https://docs.minio.io/docs/how-to-install-golang).
 
@@ -42,21 +42,18 @@ func main() {
 	useSSL := true
 
 	// Initialize minio admin client object.
-        madmClnt, err := madmin.New(endpoint, accessKeyID, secretAccessKey, useSSL)
+	madmClnt, err := madmin.New(endpoint, accessKeyID, secretAccessKey, useSSL)
 	if err != nil {
 		log.Fatalln(err)
 	}
 
 	log.Println("%v", madmClnt) // Minio admin client is now setup
-
-
+}
 ```
 
-## Quick Start Example - Service Status.
+## Quick Start Example - Server Info
 
-This example program connects to minio server, gets the current disk status.
-
-We will use the Minio server running at [https://your-minio.example.com:9000](https://your-minio.example.com:9000) in this example. Feel free to use this service for testing and development. Access credentials shown in this example are open to the public.
+This example program connects to minio server, gets the current disk status and other useful server information.
 
 #### ServiceStatus.go
 
@@ -64,6 +61,8 @@ We will use the Minio server running at [https://your-minio.example.com:9000](ht
 package main
 
 import (
+	"encoding/json"
+	"fmt"
 	"log"
 
 	"github.com/minio/minio/pkg/madmin"
@@ -81,23 +80,23 @@ func main() {
 		log.Fatalln(err)
 	}
 
-	st, err := madmClnt.ServiceStatus()
+	si, err := mdmClnt.ServerInfo()
 	if err != nil {
 		log.Fatalln(err)
 	}
-	log.Printf("%#v\n", st)
-
+	b, err := json.Marshal(si)
+	fmt.Printf("%s\n", string(b))
 }
-
 ```
+
+Replace the endpoint and access credentials above according to an actual setup.
 
 #### Run ServiceStatus
 
+The sample output below shows the result of executing the above program against a locally hosted server.
+
 ```sh
-
-go run service-status.go
-2016/12/20 16:46:01 madmin.ServiceStatusMetadata{Total:177038229504, Free:120365559808, Backend:struct { Type madmin.BackendType; OnlineDisks int; OfflineDisks int; ReadQuorum int; WriteQuorum int }{Type:1, OnlineDisks:0, OfflineDisks:0, StandardSCParity:0, RRSCParity:0}}
-
+[{"error":"","addr":"localhost:9000","data":{"storage":{"Total":460373336064,"Free":77001187328,"Backend":{"Type":2,"OnlineDisks":4,"OfflineDisks":0,"StandardSCParity":2,"RRSCParity":2}},"network":{"transferred":30599,"received":36370},"http":{"totalHEADs":{"count":0,"avgDuration":"0s"},"successHEADs":{"count":0,"avgDuration":"0s"},"totalGETs":{"count":11,"avgDuration":"0s"},"successGETs":{"count":11,"avgDuration":"0s"},"totalPUTs":{"count":0,"avgDuration":"0s"},"successPUTs":{"count":0,"avgDuration":"0s"},"totalPOSTs":{"count":0,"avgDuration":"0s"},"successPOSTs":{"count":0,"avgDuration":"0s"},"totalDELETEs":{"count":0,"avgDuration":"0s"},"successDELETEs":{"count":0,"avgDuration":"0s"}},"server":{"uptime":596915001694,"version":"2018-01-18T20:33:21Z","commitID":"e2d5a87b2676e3e01f0f4fa7ebd01205364cfb16","region":"us-east-1","sqsARN":null}}},{"error":"","addr":"minio2:9000","data":{"storage":{"Total":460373336064,"Free":77001187328,"Backend":{"Type":2,"OnlineDisks":4,"OfflineDisks":0,"StandardSCParity":2,"RRSCParity":2}},"network":{"transferred":28538,"received":11845},"http":{"totalHEADs":{"count":0,"avgDuration":"0s"},"successHEADs":{"count":0,"avgDuration":"0s"},"totalGETs":{"count":0,"avgDuration":"0s"},"successGETs":{"count":0,"avgDuration":"0s"},"totalPUTs":{"count":0,"avgDuration":"0s"},"successPUTs":{"count":0,"avgDuration":"0s"},"totalPOSTs":{"count":0,"avgDuration":"0s"},"successPOSTs":{"count":0,"avgDuration":"0s"},"totalDELETEs":{"count":0,"avgDuration":"0s"},"successDELETEs":{"count":0,"avgDuration":"0s"}},"server":{"uptime":595852367296,"version":"2018-01-18T20:33:21Z","commitID":"e2d5a87b2676e3e01f0f4fa7ebd01205364cfb16","region":"us-east-1","sqsARN":null}}},{"error":"","addr":"minio3:9000","data":{"storage":{"Total":460373336064,"Free":77001187328,"Backend":{"Type":2,"OnlineDisks":4,"OfflineDisks":0,"StandardSCParity":2,"RRSCParity":2}},"network":{"transferred":27624,"received":11708},"http":{"totalHEADs":{"count":0,"avgDuration":"0s"},"successHEADs":{"count":0,"avgDuration":"0s"},"totalGETs":{"count":0,"avgDuration":"0s"},"successGETs":{"count":0,"avgDuration":"0s"},"totalPUTs":{"count":0,"avgDuration":"0s"},"successPUTs":{"count":0,"avgDuration":"0s"},"totalPOSTs":{"count":0,"avgDuration":"0s"},"successPOSTs":{"count":0,"avgDuration":"0s"},"totalDELETEs":{"count":0,"avgDuration":"0s"},"successDELETEs":{"count":0,"avgDuration":"0s"}},"server":{"uptime":595831126778,"version":"2018-01-18T20:33:21Z","commitID":"e2d5a87b2676e3e01f0f4fa7ebd01205364cfb16","region":"us-east-1","sqsARN":null}}},{"error":"","addr":"minio4:9000","data":{"storage":{"Total":460373336064,"Free":77001187328,"Backend":{"Type":2,"OnlineDisks":4,"OfflineDisks":0,"StandardSCParity":2,"RRSCParity":2}},"network":{"transferred":27740,"received":12116},"http":{"totalHEADs":{"count":0,"avgDuration":"0s"},"successHEADs":{"count":0,"avgDuration":"0s"},"totalGETs":{"count":0,"avgDuration":"0s"},"successGETs":{"count":0,"avgDuration":"0s"},"totalPUTs":{"count":0,"avgDuration":"0s"},"successPUTs":{"count":0,"avgDuration":"0s"},"totalPOSTs":{"count":0,"avgDuration":"0s"},"successPOSTs":{"count":0,"avgDuration":"0s"},"totalDELETEs":{"count":0,"avgDuration":"0s"},"successDELETEs":{"count":0,"avgDuration":"0s"}},"server":{"uptime":595349958375,"version":"2018-01-18T20:33:21Z","commitID":"e2d5a87b2676e3e01f0f4fa7ebd01205364cfb16","region":"us-east-1","sqsARN":null}}}]
 ```
 
 ## API Reference
@@ -119,4 +118,3 @@ go run service-status.go
 ## Contribute
 
 [Contributors Guide](https://github.com/minio/minio/blob/master/CONTRIBUTING.md)
-

